src/django_components/component.py

```diff
@@ -303,135 +320,195 @@ def get_context_data(self):
 
         As the `{{ data.hello }}` is taken from the "provider".
         """
-        comp_name = self.registered_name or self.__class__.__name__
         if self._context is None:
             raise RuntimeError(
-                f"Method 'inject()' of component '{comp_name}' was called outside of 'get_context_data()'"
+                f"Method 'inject()' of component '{self.name}' was called outside of 'get_context_data()'"
             )
 
-        return get_injected_context_var(comp_name, self._context, key, default)
+        return get_injected_context_var(self.name, self._context, key, default)
+
+    @classmethod
+    def render_to_response(
```

Comment:
`render_to_response` is now also a class method

Reply:
This breaks backwards compatibility if people have self.render_to_response I think? Or can class methods be called on instances?